### PR TITLE
[performance](load) support parallel memtable flush for unique key tables

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

Assign a `segment_id` to `MemTable` early before submit flush task, and exec the flush tasks in parallel.

In this way, we can improve the stream load peformance of 1 tablet unique key tables by around 100%.

Stream load 7.5G tpch lineitem table into a 1 tablet unique key table (3 BE, 16C 64G):

| replicas | load time (before) | load time (after) |
| --- | --- | --- |
| 1 | 118s | 48s |
| 3 | 161s | 60s |

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

File: be/src/olap/delta_writer.cpp [hunk 1]
```diff
@@ -209,9 +209,9 @@ Status DeltaWriter::init() {
     _reset_mem_table();
 
     // create flush handler
-    // unique key should flush serial because we need to make sure same key should sort
-    // in the same order in all replica.
-    bool should_serial = _tablet->keys_type() == KeysType::UNIQUE_KEYS;
+    // by assigning seq_id to memtable before submiting to flush executor,
+    // we can make sure same keys sort in the same order in all replicas.
+    bool should_serial = false;
```
[the remark this answers is not in the document]
Comment (kaijchen) [reply be/src/olap/delta_writer.cpp:214]: I keep this for docs and easy rollback. The compiler will optimize it.
We can change it later when this feature is stable, probably change the signature of `create_flush_token` as well.

**Pull request merged**